rockcraft/pebble.py

```diff
@@ -69,10 +69,7 @@ def define_pebble_layer(
         )
 
         new_layer_prefix = f"{(int(prefixes[-1]) + 1):03}" if prefixes else "001"
-        # TODO: we need proper name validation as Pebble is quite unforgiving
```

Comment:
In the end I followed pebble's direction here but we should consider the `_` case: if it's too common maybe we should still support it?